extensions/3rdparty/ArrayExtension/ArrayExtension.php

```diff
@@ -575,7 +575,7 @@ function arrayunion( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
 *    see: http://www.php.net/manual/en/function.array-diff.php
 */
 
-    function arraydiff( &$parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
+    function arraydiff( $parser, $arrayid_new , $arrayid1 , $arrayid2 ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)